src/content_cao.cpp

```diff
@@ -1302,14 +1308,17 @@ void GenericCAO::updateTexturePos()
 	}
 }
 
-void GenericCAO::updateTextures(const std::string &mod)
```

Comment:
Argh 
You're right of course if mod is a reference it's actually the identical variable which is used on calling the function. As I do set this variable in the function itself the parameter is modified too. So I'd have to create a copy of it to be sure to be able to set the previous way in all cases and making this function call by reference wouldn't avoid any copy.